src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/WindowsRuntime/Windows/UI/ViewManagement/InputPane.cs

```diff
@@ -65,18 +60,17 @@ static InputPane()
                 // We don't want to throw here - so wrap in try..catch
                 try
                 {
-                    s_WinRTType = Type.GetType(s_TypeName);
-
                     // If we cannot get a new activation factory, then we cannot support
                     // this platform.  As such, null out the type to guard instantiations.
                     if (GetWinRtActivationFactory(forceInitialization: true) == null)
                     {
-                        s_WinRTType = null;
+                        _isSupported = false;
                     }
+                    _isSupported = true;
```

Comment:
This looks wrong. If `GetWinRtActivationFactory` returns `null`, then the code first assigns `_isSupported = false` but immediately overwrites that with `_isSupported = true`.

Reply:
So, if `GetWinRtActivationFactory` returns `null`, then `_isSupported` ends up being `true` even though InputPane is not actually supported. That then causes `private InputPane(IntPtr? hwnd)` constructor to call `GetWinRtActivationFactory` again, instead of throwing `PlatformNotSupportedException` right away. However, that call presumably returns `null` again, and the constructor then throws `PlatformNotSupportedException` anyway. I think this bug will not cause any real problem then, although it may cause `RoGetActivationFactory` to be unnecessarily retried and thus perhaps slow things down a bit.